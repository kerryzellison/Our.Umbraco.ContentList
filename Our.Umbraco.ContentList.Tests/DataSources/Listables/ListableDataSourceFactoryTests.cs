using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.Windows;
using NUnit.Framework;
using Our.Umbraco.ContentList.DataSources.Listables;
using Our.Umbraco.ContentList.Install;
using Umbraco.Tests.Testing;

namespace Our.Umbraco.ContentList.Tests.DataSources.Listables
{
    [TestFixture]
    [UmbracoTest]
    [UseReporter(typeof(VisualStudioReporter))]
    public class ListableDataSourceFactoryTests : UmbracoTestBase
    {
        protected override void Compose()
        {
            base.Compose();

            new ListableDataSourcesComposer().Compose(Composition);
        }

        [Test]
        public void Finds_Metadata_For_Data_Source()
        {
            var parameters = ListableDataSourceFactory.CreateParameters(typeof(ListableChildrenDataSource));
            Approvals.VerifyJson(parameters.ToJson());
        }

        [Test]
        public void Has_List_Of_DataSources_With_Metadata()
        {
            var datasources = ListableDataSourceFactory.GetDataSources();
            Assert.AreNotEqual(0, datasources.Count);
            Console.WriteLine(datasources.ToJson(true));
            Approvals.VerifyJson(datasources.ToJson());
        }

        [Test]
        public void Creates_Datasource()
        {
            var fullName = typeof(ListableChildrenDataSource).FullName;

            var datasource = new ListableDataSourceFactory().Create(fullName);

            Assert.IsInstanceOf<ListableChildrenDataSource>(datasource);
        }
    }
}
