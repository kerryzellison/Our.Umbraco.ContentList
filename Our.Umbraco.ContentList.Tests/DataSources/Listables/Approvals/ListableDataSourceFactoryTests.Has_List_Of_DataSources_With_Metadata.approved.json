[
    {
        "key": "Our.Umbraco.ContentList.DataSources.Listables.ListablesByXPathDataSource, Our.Umbraco.ContentList",
        "name": "XPath Query",
        "parameters": [
            {
                "label": "Query",
                "key": "xpath",
                "view": "/Umbraco/views/propertyeditors/textbox/textbox.html",
                "config": null
            },
            {
                "label": "Sort by",
                "key": "sort",
                "view": "/App_Plugins/Our.Umbraco.ContentList/propertyeditors/dropdown/dropdown.html",
                "config": {
                    "items": [
                        {
                            "id": "sortorder",
                            "value": "Sort order"
                        },
                        {
                            "id": "dateasc",
                            "value": "Date"
                        },
                        {
                            "id": "datedesc",
                            "value": "Date descending"
                        }
                    ]
                }
            }
        ]
    },
    {
        "key": "Our.Umbraco.ContentList.DataSources.Listables.ListablesByLuceneDataSource, Our.Umbraco.ContentList",
        "name": "Lucene Query",
        "parameters": [
            {
                "label": "Query",
                "key": "query",
                "view": "/Umbraco/views/propertyeditors/textbox/textbox.html",
                "config": null
            }
        ]
    },
    {
        "key": "Our.Umbraco.ContentList.DataSources.Listables.ListableChildrenDataSource, Our.Umbraco.ContentList",
        "name": "List of children",
        "parameters": [
            {
                "label": "Sort by",
                "key": "sort",
                "view": "/App_Plugins/Our.Umbraco.ContentList/propertyeditors/dropdown/dropdown.html",
                "config": {
                    "items": [
                        {
                            "id": "sortorder",
                            "value": "Sort order"
                        },
                        {
                            "id": "dateasc",
                            "value": "Date"
                        },
                        {
                            "id": "datedesc",
                            "value": "Date descending"
                        }
                    ]
                }
            }
        ]
    },
    {
        "key": "Our.Umbraco.ContentList.DataSources.Listables.ListableByNodeDataSource, Our.Umbraco.ContentList",
        "name": "List of items from tree",
        "parameters": [
            {
                "label": "From node",
                "key": "parentNode",
                "view": "/Umbraco/Views/prevalueeditors/treepicker.html",
                "config": null
            },
            {
                "label": "Sort by",
                "key": "sort",
                "view": "/App_Plugins/Our.Umbraco.ContentList/propertyeditors/dropdown/dropdown.html",
                "config": {
                    "items": [
                        {
                            "id": "sortorder",
                            "value": "Sort order"
                        },
                        {
                            "id": "dateasc",
                            "value": "Date"
                        },
                        {
                            "id": "datedesc",
                            "value": "Date descending"
                        }
                    ]
                }
            }
        ]
    }
]